src/main/java/org/skriptlang/skript/lang/entry/EntryContainer.java

```diff
@@ -170,4 +171,20 @@ public Object getOptional(String key, boolean useDefaultValue) {
 		return value;
 	}
 
+	/**
+	 * Check to see if an entry data with the key matching {@code key} exists.
+	 * @param key The key to check
+	 * @return true if an entry data with the matching key exists
+	 */
+	public boolean hasEntry(@NotNull String key) {
+		if (entryValidator == null || handledNodes == null)
+			return false;
+
+		for (EntryData<?> entryData : entryValidator.getEntryData()) {
+			if (entryData.getKey().equalsIgnoreCase(key))
+				return true;
+		}
+		return false;
```

Comment:
After speaking with the original issue author on Discord, they clarified that they want a method that checks whether an entry was used at runtime (that is, a node in a script matched with a key). Thus, you should be able to simply use:
```suggestion
		return handledNodes.containsKey(key);
```

For example, if you have:
```
command /test:
    trigger:
        ...
```
If you did `entryContainer.hasEntry("permission")`, it should return false since it wasn't used in the code (even though an EntryData with the key "permission" exists on the Command EntryValidator). The documentation should also be updated to reflect this behavior.

Reply:
So I was wrong to second guess myself on if `handledNodes` would of worked, god I hate myself, anyways thanks pickle for double checking with me.